packages/flutter/test/widgets/snapshot_widget_test.dart

```diff
@@ -261,6 +263,39 @@ void main() {
     expect(tester.takeException(), isNull);
     expect(tester.layers.last, isA<PlatformViewLayer>());
   }, skip: kIsWeb); // TODO(jonahwilliams): https://github.com/flutter/flutter/issues/106689
+
+  testWidgets('SnapshotWidget should have same result when enabled', (WidgetTester tester) async {
+    binding.window
+      ..physicalSizeTestValue = const Size(10, 10)
+      ..devicePixelRatioTestValue = 1;
+    addTearDown(() => binding.window
```

Comment:
I think we normally use `try { } finally { }` for these. Maybe we should use addTearDown if its easier

Reply:
Depends on you, if you like I can refactor to try-finally

In addition, I personally have a utility function internally, so I just write

```
binding.window.setUpTearDown(physicalSizeTestValue: ..., )
```

and it auto setup and teardown

if you like I can PR that in a separate PR

Reply:
I think the addTearDown is fine!